### PR TITLE
Release Godot 4.1.2 and publish release notes

### PR DESCRIPTION
**Pull request by YuriSizov**
Ready for proofreading and review, but drafting for now because I need to update the interactive changelog and fetch some numbers from it.

### CODE REVIEW

File: collections/_article/maintenance-release-godot-4-1-2.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+Such is the case of Godot 4.1.2, which is officially available starting today! Engine contributors provided fixes for almost every area of the engine, so make sure to read the changelog [below](#changes). Here's a short summary of some of the more interesting changes.
+
+- A number of performance issues have been addressed in rendering ([GH-80070](https://github.com/godotengine/godot/pull/80070), [GH-80407](https://github.com/godotengine/godot/pull/80407)), GUI ([GH-79325](https://github.com/godotengine/godot/pull/79325), [GH-80857](https://github.com/godotengine/godot/pull/80857)), and engine core ([GH-81037](https://github.com/godotengine/godot/pull/81037)). The latter fix should also improve memory profile of the engine if you instance a lot of plain objects.
```
Comment (JohnVeness) [inline collections/_article/maintenance-release-godot-4-1-2.md]:
```suggestion
- A number of performance issues have been addressed in rendering ([GH-80070](https://github.com/godotengine/godot/pull/80070), [GH-80407](https://github.com/godotengine/godot/pull/80407)), GUI ([GH-79325](https://github.com/godotengine/godot/pull/79325), [GH-80857](https://github.com/godotengine/godot/pull/80857)), and engine core ([GH-81037](https://github.com/godotengine/godot/pull/81037)). The latter fix should also improve the memory profile of the engine if you instance a lot of plain objects.
```

File: collections/_article/maintenance-release-godot-4-1-2.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+excerpt: "As Godot 4.2 enters feature freeze this week, a new Godot 4.1 maintenance release arrives with a handful of improvements to stability and performance!"
+categories: ["release"]
+author: Yuri Sizov
+image: /storage/blog/covers/maintenance-release-godot-4-1-2.webp
+image_caption_title: Robot Detour
+image_caption_description: A game by Nozomu Games
+date: 2023-10-04 14:00:00
+---
+
+Godot 4.1 is celebrating its [3 month anniversary](/article/godot-4-1-is-here) this week, and that means Godot 4.2 is entering the beta testing phase shortly after. Each development cycle is fruitful with bugfixes and other sorts of improvements, and it's an integral part of our development process to identify such changes. If these contributions prove to be safe and stable, we compile a bunch of them into a patch release for the current stable version of the engine.
```
Comment (clayjohn) [inline collections/_article/maintenance-release-godot-4-1-2.md]: Just a grammar and style suggestion that simplifies things a bit but gets the same point across. 

```suggestion
Godot 4.1 is celebrating its [3 month anniversary](/article/godot-4-1-is-here) this week, and that means Godot 4.2 is entering the beta testing phase shortly after. This development cycle has been packed with bugfixes and other  improvements. As always, we take the safest of these contributions and we compile them into a patch release for the current stable version of the engine.
```

**Review by clayjohn — COMMENTED** [review]
Looks great! I added one grammar/style suggestion

File: collections/_article/maintenance-release-godot-4-1-2.md (hunk 2 above)
Comment (YuriSizov) [reply collections/_article/maintenance-release-godot-4-1-2.md]: Thanks, that's much better!

File: collections/_article/maintenance-release-godot-4-1-2.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+- Another critical platform-specific bug has been identified for the web platform, with file system operations failing and causing issues for developers and users of Godot's own online editor. It is now fixed ([GH-79866](https://github.com/godotengine/godot/pull/79866)), which means you can once again save your projects while working on the web.
+
+- As many game developers compile Godot and custom forks by themselves, it's important for us to ensure buildsystem stability and flexibility as well as that of the rest of the engine — even for previous releases. This version includes a series of platform fixes, targeting XCode 15 ([GH-82458](https://github.com/godotengine/godot/pull/82458)), Visual Studio 2022 for macOS ([GH-81802](https://github.com/godotengine/godot/pull/81802)), as well as general Visual Studio and MSVC improvements ([GH-79238](https://github.com/godotengine/godot/pull/79238)).
```
Comment (akien-mga) [inline collections/_article/maintenance-release-godot-4-1-2.md]: Technically, the VS for Mac change is not for compiling Godot, but for using it as external IDE to edit and compile C# scripts. But I don't think it a big deal to keep it as is either.

File: collections/_article/maintenance-release-godot-4-1-2.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+- Rendering: Vulkan: Fix texture update ([GH-80781](https://github.com/godotengine/godot/pull/80781)).
+- Rendering: Vulkan: Fix crash with many Omni/SpotLights, Decals or ReflectionProbes ([GH-80845](https://github.com/godotengine/godot/pull/80845)).
+- Shaders: Allow more hint types for uniform arrays ([GH-79100](https://github.com/godotengine/godot/pull/79100)).
+- Shaders: Fix shader type detection ([GH-79287](https://github.com/godotengine/godot/pull/79287)).
+- Shaders: Fix Shader and ShaderInclude resource loading ([GH-80705](https://github.com/godotengine/godot/pull/80705)).
+- XR: Fix issue with accessing hand tracking without timing info ([GH-78817](https://github.com/godotengine/godot/pull/78817)).
+- XR: Ensure OpenXR classes are declared properly ([GH-81037](https://github.com/godotengine/godot/pull/81037)).
+- Thirdparty: FreeType 2.13.2, ICU4C 73.2, libpng 1.6.40, libwebp 1.3.2, mbedtls 2.28.4, miniupnpc 2.2.5, openxr 1.0.28, tinyexr 1.0.7.
+
+- As well as many improvements to the documentation.
```
Comment (akien-mga) [inline collections/_article/maintenance-release-godot-4-1-2.md:108]: How does it render with a separation line? Two lists?

**Review by akien-mga — APPROVED** [review]

File: collections/_article/maintenance-release-godot-4-1-2.md (hunk 4 above)
Comment (YuriSizov) [reply collections/_article/maintenance-release-godot-4-1-2.md:108]: No, it's still one list. Just helps visually when editing.

File: collections/_article/maintenance-release-godot-4-1-2.md (hunk 3 above)
Comment (YuriSizov) [reply collections/_article/maintenance-release-godot-4-1-2.md]: Ah, so it shouldn't have been marked as buildsystem then.

**Comment by YuriSizov** [comment]
Thanks for reviews!

**Pull request merged**